File: Dao/pedidos-rest-24-main/pedidos-rest-24-main/src/main/java/es/uvigo/mei/pedidos/servicios/TrabajadorService.java
package es.uvigo.mei.pedidos.servicios;

import java.util.List;
import java.util.Optional;

import es.uvigo.mei.pedidos.entidades.Trabajador;

public interface TrabajadorService {
    public Trabajador crear(Trabajador trabajador);
	public Trabajador modificar(Trabajador trabajador);
	public void eliminar(Trabajador trabajador);
	public List<Trabajador> buscarPorDNI(String dni);
	public List<Trabajador> buscarTodos();
	public List<Trabajador> buscarPorNombre(String nombre);
	public List<Trabajador> buscarPorBrigada(long  id);
	public List<Trabajador> buscarPorBrigadaId(long id);
	public List<Trabajador> buscarPorContrto(long id);
	public Optional<Trabajador> buscarPorId(long  id);

    
}

File: Dao/pedidos-react-23-main/pedidos-react-23-main/src/services/trabajadorService.js
import api from "./pedidosAPI";

class Trabajadoreservice {
  buscarTodas() {
    return api.get("/trabajadores");
  }
  buscarPorDescripcion(descripcion) {
    return api.get(`/trabajadores?descripcion=${descripcion}`);
  }

  buscarPorId(id) {
    return api.get(`/trabajadores/${id}`);
  }

  buscarPorDNI(dni) {
    return api.get(`/trabajadores?dni=${dni}`);
  }
  buscarPorNombre(nombre) {
    return api.get(`/trabajadores?nombre=${nombre}`);
  }

  buscarPorbrigadaId(id) {
    return api.get(`/trabajadores/brigada/${id}`);
  }

  buscarPorcontratoId(id) {
    return api.get(`/trabajadores/contrato/${id}`);
  }

  getBrigada(id) {   
    return api.get(`/trabajadores/${id}/brigada`);
  }
  crear(data) {
    return api.post("/trabajadores", data);
  }

  modificar(id, data) {
    return api.put(`/trabajadores/${id}`, data);
  }

  eliminar(id) {
    return api.delete(`/trabajadores/${id}`);
  }

 
}

const service = new Trabajadoreservice();
export default service 
